tests/readrecursive.c

```diff
@@ -68,7 +68,7 @@ static int readdirectory(char *dirname, int *dirsread, int *filesread, int *erro
         else {
             int bytes_read;
             char rbuf [1025];
-            fd = open(fn, O_RDWR);
+            fd = open(fn, O_RDONLY);
```

Comment:
I don't know this code well enough to understand what the change here is for, or why we need it?